java/HealthVault Mobile/HVSample/src/main/java/com/microsoft/hsg/android/hvsample/MainActivity.java

```diff
@@ -193,76 +189,25 @@ public void onListItemClick(ListView parent, View v, int position, long id) {
         }
     }
 	
-	/*private void registerHandlers() {
-
-		// Connect button
-		Button connect = (Button) findViewById(R.id.connect);
-		connect.setOnClickListener(new View.OnClickListener() {
-            public void onClick(View view) {
-            	doConnect();
-            }
-        });
-		
-		Button weightsBtn = (Button) findViewById(R.id.weights);
-		weightsBtn.setOnClickListener(new View.OnClickListener() {
-            public void onClick(View view) {
-            	if (service.isAppConnected()) {
-            		Intent intent = new Intent(MainActivity.this, WeightActivity.class);
-                    startActivity(intent);
-            	}
-            }
-        });
-		
-		Button fileUploadBtn = (Button) findViewById(R.id.fileUpload);
-		fileUploadBtn.setOnClickListener(new View.OnClickListener() {
-            @SuppressWarnings("unchecked")
-			public void onClick(View view) {
-            	if (service.isAppConnected()) {
-            		String filename = writeFile();
-            		
-            		InputStream source;
-					try {
-						source = openFileInput(filename);
-						
-						File hvFile = new File();
-						hvFile.setName(filename);
-						
-						hvClient.start();
-	            		
-	            		hvClient.asyncRequest(hvFile.uploadAsync(((HVApp)getApplication()).getCurrentRecord(), null, source), 
-	            				new MainActivityCallback(MainActivityCallback.UpdateRecords));
-					} catch (Exception e) {
-						// TODO Auto-generated catch block
-						e.printStackTrace();
-					}
-            	}
-            }
-        });
-		
-		Button vocabTest = (Button) findViewById(R.id.vocabTest);
-		vocabTest.setOnClickListener(new View.OnClickListener() {
-            public void onClick(View view) {
-            	Intent intent  = new Intent(MainActivity.this, VocabActivity.class);
-            	startActivity(intent);
-            }
-        });
-	}*/
-
 	private String writeFile() {
 		String filename = "writefile" + (int)(Math.random() * 100)  + ".txt";
 		String string = "This is from file upload";
-		FileOutputStream outputStream;
+		FileOutputStream outputStream = null;
 
 		try {
 		  outputStream = openFileOutput(filename, Context.MODE_PRIVATE);
 		  outputStream.write(string.getBytes());
-		  outputStream.close();
-		  
+
 		  return filename;
 		} catch (Exception e) {
 		  e.printStackTrace();
+		} finally {
+			try {
+				outputStream.close();
```

Comment:
null check first, if the openFileOutput call throws, outputStream will be null and this will crash.